app/controllers/assessments_controller.rb

```diff
@@ -70,10 +70,14 @@ def new
     @threshold_hash = jurisdiction.hierarchical_symptomatic_condition.threshold_condition_hash
     @translations = @assessment.translations
     @contact_info = jurisdiction.contact_info
-    @lang = permitted_params[:lang] if %w[en es es-PR so fr].include?(params[:lang])
-    @lang = 'en' if @lang.nil? # Default to english
+
+    @lang = LegacyLanguages.legacy_language_code?(permitted_params[:lang]) ?
+            LegacyLanguages.translate_legacy_language_code(permitted_params[:lang]) : permitted_params[:lang]
+    @lang = 'eng' unless Languages.supported_language?(@lang)
```

Comment:
it's probably more performant and safer to just check for `@lang.nil?`

Reply:
Im not sure i understand what you're saying here. This logic seems to be correct to me.

Reply:
after looking through it again, it looks good to me! I was thinking there might be an edge case somehow where `@lang` would somehow never be set because this condition would be satisfied, but that would only ever happen if it's valid